azurerm/internal/services/web/resource_arm_app_service_hybrid_connection.go

```diff
@@ -0,0 +1,205 @@
+			"relay_arm_uri": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+			"namespace_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"relay_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"hostname": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+			"port": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				ValidateFunc: validate.PortNumberOrZero,
+			},
+			"service_bus_namespace": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringMatch(
+					regexp.MustCompile("^[a-zA-Z][-a-zA-Z0-9]{0,100}[a-zA-Z0-9]$"),
+					"The namespace can contain only letters, numbers, and hyphens. The namespace must start with a letter, and it must end with a letter or number.",
+				),
+			},
+			"service_bus_suffix": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+			"send_key_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+			"send_key_value": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Sensitive:    true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+		},
+	}
+}
+
+func resourceArmAppServiceHybridConnectionCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Web.AppServicesClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("app_service_name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	relayArmURI := d.Get("relay_arm_uri").(string)
+	relayId, err := azure.ParseAzureResourceID(relayArmURI)
+	if err != nil {
+		return fmt.Errorf("Error parsing relay_arm_uri as ID: %s", relayArmURI)
+	}
+
+	namespaceName := relayId.Path["namespaces"]
+	if namespaceName == "" {
+		return fmt.Errorf("Failed to parse relay namespace from ID: %s", relayArmURI)
+	}
+
+	relayName := relayId.Path["hybridConnections"]
+	if relayName == "" {
+		return fmt.Errorf("Failed to parse relay name from ID: %s", relayArmURI)
+	}
+
+	port := int32(d.Get("port").(int))
+
+	hybridConnectionProperties := web.HybridConnectionProperties{
+		ServiceBusNamespace: utils.String(d.Get("service_bus_namespace").(string)),
+		RelayName:           &relayName,
+		RelayArmURI:         &relayArmURI,
+		Hostname:            utils.String(d.Get("hostname").(string)),
+		Port:                &port,
+		SendKeyName:         utils.String(d.Get("send_key_name").(string)),
+		SendKeyValue:        utils.String(d.Get("send_key_value").(string)),
+		ServiceBusSuffix:    utils.String(d.Get("service_bus_suffix").(string)),
+	}
+
+	connectionEnvelope := web.HybridConnection{}
+	connectionEnvelope.HybridConnectionProperties = &hybridConnectionProperties
```

Comment:
This could still become
```
	connectionEnvelope := web.HybridConnection{
		HybridConnectionProperties = &hybridConnectionProperties
```

or more consistent with the rest of the provider:

```
	connectionEnvelope := web.HybridConnection{
		HybridConnectionProperties = &web.HybridConnectionProperties{
		ServiceBusNamespace: utils.String(d.Get("service_bus_namespace").(string)),
...
```